src/executors/collect_battery.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Created by xiaofei.lan@rokid.com on 2019/04/26.
+ */
+
+#include "collect_battery.h"
+
+static void readFile(const char *path, char *buf, int *value) {
+  FILE *file;
+  int size = 9;
+  file = fopen(path, "rb");
+  if (file == NULL)
+  {
+    YODA_SIXSIX_FLOG("No such file: %s. skip collect info.", path);
+    if (value != NULL)
+    {
+      *value = 0;
+    }
+    return;
+  }
+  size = fread(buf, 1, size, file);
```

Comment:
`fread` returns the number of objects read successfully, you should handle the return value which is not equal with SIZE, here's an example:
```C
    size_t ret_code = fread(b, 1, SIZE, fp); // reads an array of doubles
    if(ret_code == SIZE) {
        puts("Array read successfully, contents: ");
    } else { // error handling
       if (feof(fp)) // end of file
          printf("Error reading test.bin: unexpected end of file\n");
       else if (ferror(fp)) {
           perror("Error reading test.bin");
       }
    }
```

Reply:
yes. I have catch in case:
```c
if (size <= 0)
{
    if (value != NULL) *value = 0;
    return;
}
```
The length of file is uncertain, so i can't promise the value of size is always be equal SIZE.